### PR TITLE
Fix bug in INSTALL_METHOD

### PR DESCRIPTION
**Pull request by markuspf**
GET_OPER_FLAGS returns `fail` not `false` if the operation is not known.

### CODE REVIEW

**Comment by fingolfin** [comment]
Shouldn't this go to stable-4.9?

File: hpcgap/lib/oper1.g [hunk 1]
```diff
@@ -462,7 +462,7 @@ BIND_GLOBAL( "INSTALL_METHOD",
 
       # find the operation
       req := GET_OPER_FLAGS(opr);
-      if req = false  then
+      if req = fail  then
```
Comment (fingolfin) [inline hpcgap/lib/oper1.g]: Good catch, this slipped with b1a0e86e43e11bab71d2aef89fb54529f2fe4f24 from PR #1707 by @frankluebeck and I failed to notice it during review :-(

**Comment by frankluebeck** [comment]
Thanks for catching this. Of course, this is only detected when `InstallMethod` is called with invalid arguments.  Would it be useful to add a test like this:
```
gap> InstallMethod(Integers, [IsInt], function(n) return n; end);
Error, <opr> is not an operation
gap> InstallMethod(Size, [IsInt], function(n) return n; end);
Error, required filters [ "IsInt", "IsRat", "IsCyc", "IsExtAElement", 
  "IsNearAdditiveElement", "IsNearAdditiveElementWithZero", 
  "IsNearAdditiveElementWithInverse", "IsAdditiveElement", "IsExtLElement", 
  "IsExtRElement", "IsMultiplicativeElement", "IsMultiplicativeElementWithOne"
    , "IsMultiplicativeElementWithInverse", "IsZDFRE", "IsAssociativeElement",
  "IsAdditivelyCommutativeElement", "IsCommutativeElement", "IsCyclotomic" ]
for 1st argument do not match a declaration of Size
```

**Comment by fingolfin** [comment]
This PR updates `oper1.g` which is compiled. Thus `src/c_oper1.c` and `hpcap/src/c_oper1.c` must be updated, too (by running `make docomp` twice in a GAP and a HPC-GAP build).

**Comment by markuspf** [comment]
Cleaned up, test added. 

Slightly unrelated note: there are more tests coming for the whole `Declare`, `New`, `Install` machinery for operations/methods.

I have no idea why it rearranges `c_type1.c` in `make docomp` for HPC-GAP.

**Pull request merged**